core/trino-spi/src/main/java/io/trino/spi/eventlistener/QueryMetadata.java

```diff
@@ -56,6 +57,7 @@ public QueryMetadata(
             List<RoutineInfo> routines,
             URI uri,
             Optional<String> plan,
+            Optional<String> anonymizedJsonPlan,
```

Comment:
What's the use case for including both the regular plan and the anonymized plan? Most event listeners will just forward and store the event as is, which defeats the purpose of having an anonymized plan.

Reply:
It is possible to have different use cases for regular plan and anonymised plan. E.g. Regular plan can be used to display query plan for past queries to a user in a UI. This would be user specific data that might not be made easily accessible to everyone, it's also formatted in a way that is not ideal for offline analysis.
Anonymized plan could be sent to a different downstream system which is more widely accessible and easier to work with for offline analysis.
Currently we send same event to all event listeners, so we also can't use different event listeners to get different versions of plan.